File: openapi-generator/kotlin-spring/src/main/kotlin/org/openapitools/model/Sensor.kt
package org.openapitools.model

import java.util.Objects
import com.fasterxml.jackson.annotation.JsonProperty
import javax.validation.Valid
import javax.validation.constraints.*

/**
 * Contains information about a sensor.
 * @param id ID of the sensor.
 * @param macAddress MAC address of the sensor.
 * @param name Name of the sensor.
 */
data class Sensor (

        @get:NotNull 
        @JsonProperty("id") val id: Long,

        @JsonProperty("macAddress") val macAddress: String? = null,

        @JsonProperty("name") val name: String? = null
) {

}

File: openapi-generator/javascript-closure-angular/API/Client/Sensor.js
goog.provide('API.Client.Sensor');

/**
 * Contains information about a sensor.
 * @record
 */
API.Client.Sensor = function() {}

/**
 * ID of the sensor.
 * @type {!number}
 * @export
 */
API.Client.Sensor.prototype.id;

/**
 * MAC address of the sensor.
 * @type {!string}
 * @export
 */
API.Client.Sensor.prototype.macAddress;

/**
 * Name of the sensor.
 * @type {!string}
 * @export
 */
API.Client.Sensor.prototype.name;

